Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
+{
+    [SerializeField]
+    private int width;
+
+    [SerializeField]
+    private int height;
+
+    public Transform brick;
+    private int[,] Maze;
+
+    private enum TileType { path, wall };
+    private List<Vector3> pathMazes = new List<Vector3>();
+    private Stack<Vector2> tileStack = new Stack<Vector2>();
+
+    private readonly List<Vector2> offsets = new List<Vector2>
+    {
+        new Vector2(0, 1),
+        new Vector2(0, -1),
+        new Vector2(1, 0),
+        new Vector2(-1, 0)
+    };
+
+    private System.Random rnd = new System.Random();
```

Comment:
are you familiar with seeds for random? If not, stop by and ask. They could add a little extra to this game but definitely aren't necessary